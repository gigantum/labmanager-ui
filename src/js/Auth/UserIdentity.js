//vendor
import {
  graphql,
} from 'react-relay'
//mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const userIdentityQuery = graphql`
  query UserIdentityQuery{
    userIdentity{
      id
      username
      email
      givenName
      familyName
      isSessionValid
    }
  }
`;

const UserIdentity = {
  getUserIdentity: () => {

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(userIdentityQuery(), {}).then((response, error) => {

          if(response){
            resolve(response)
          }else{

            reject(response)
          }
        }).catch((error) =>{

          reject(error)
        })
      }

      fetchData()
    })
  },
  removeUserIdentity: () => {
    RemoveUserIdentityMutation(()=>{
      console.log('removed user')
    })
  }

}

export default UserIdentity
