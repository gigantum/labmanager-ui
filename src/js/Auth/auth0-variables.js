export const AUTH_CONFIG = {
  domain: 'calum.auth0.com',
  clientId: '9wFN5CZA4Z9xa1r8KZIZbevpjHG7bAab',
  callbackUrl: window.location.origin + '/callback'
}
