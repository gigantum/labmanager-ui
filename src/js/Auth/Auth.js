import history from 'JS/history';
import auth0 from 'auth0-js';
import { AUTH_CONFIG } from './auth0-variables';
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//store
import store from 'JS/redux/store'


export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience:  AUTH_CONFIG.audience,
    responseType: 'token id_token',
    scope: 'openid profile email user_metadata'
  });

  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.handleAuthentication = this.handleAuthentication.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);

  }

  login() {
    this.auth0.authorize();
  }

  handleAuthentication() {

    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {

        this.setSession(authResult);

      } else if (err) {
        console.error(err);
      //  alert(`Error: ${err.error}. Check the console for further details.`);
      }
    });
  }

  setSession(authResult) {

    // Set the time that the access token will expire at
    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
    localStorage.setItem('access_token', authResult.accessToken);
    localStorage.setItem('id_token', authResult.idToken);
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('family_name', authResult.idTokenPayload.family_name);
    localStorage.setItem('given_name', authResult.idTokenPayload.given_name);
    localStorage.setItem('email', authResult.idTokenPayload.email);
    localStorage.setItem('username', authResult.idTokenPayload.nickname);
    //redirect to labbooks when user logs in

    let storrageRoute = sessionStorage.getItem('CALLBACK_ROUTE')
    let route = storrageRoute !== '' ? storrageRoute : `/labbooks`

    history.replace(route)
  }

  logout() {

    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('family_name')
    localStorage.removeItem('given_name');
    localStorage.removeItem('email');
    localStorage.removeItem('username');
    RemoveUserIdentityMutation(()=>{
      //redirect to root when user logs out
      history.replace('/');
    })
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    let expiresAt = JSON.parse(localStorage.getItem('expires_at'));

    return new Date().getTime() < expiresAt;
  }
}
