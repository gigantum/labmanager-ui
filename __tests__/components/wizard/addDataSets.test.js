import React from 'react';
import AddDataSets from 'Components/wizard/AddDataSets';
import {mount} from 'enzyme'
import renderer from 'react-test-renderer';
import Auth from 'JS/Auth/Auth';

let toggleDisabledContinue = () => {}
let setComponent = () => {}

test('Test AddDataSets rendering', () => {

  const component = renderer.create(
    <AddDataSets
      toggleDisabledContinue={toggleDisabledContinue}
      setComponent={setComponent}
      nextWindow={'addDatasets'}
      labbookName={'demo-lab-book'}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
