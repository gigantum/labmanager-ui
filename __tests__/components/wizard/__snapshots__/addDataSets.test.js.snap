// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddDataSets rendering 1`] = `
<div
  className="AddDatasets flex flex--column justify--space-around"
>
  <p>
    Select a Dataset
  </p>
  <div
    className="AddDatasets__selected-items flex justify-center"
  />
  <div
    className="AddDatasets__selection flex justify-center"
  />
</div>
`;
